server/gc/fsck.c

```diff
@@ -26,20 +66,27 @@ check_blocks (SeafFSManager *mgr, FsckRes *res, const char *file_id)
 
     seafile = seaf_fs_manager_get_seafile (mgr, repo->store_id, repo->version, file_id);
     if (!seafile) {
-        seaf_warning ("Failed to find file %s.\n", file_id);
+        if (fsck_data->clean) {
+            seaf_message ("Remove curropted file %.8s.\n", file_id);
+            seaf_fs_manager_delete_object (mgr, repo->store_id, repo->version, file_id);
```

Comment:
This is dangerous! You may remove a valid object when it's temporarily not accessible, e.g. I/O error.